cdap-ui/app/cdap/components/FieldLevelLineage/v2/Context/FllContext.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+import React from 'react';
+import data from './sample_response';
+import uuidV4 from 'uuid/v4';
+
+interface INode {
+  id: string;
+  name: string;
+  group: number;
+}
+
+const FllContext = React.createContext({});
+
+function makeNodes(fields) {
+  const targetNodes = fields.map((field) => {
+    const id = uuidV4();
+    return {
+      id,
+      name: field,
+      group: data.entityId.dataset,
+    };
+  });
+  return targetNodes;
+}
+
+function getFieldsAndLinks(d) {
+  const incoming = parseRelations(d.incoming);
+  const outgoing = parseRelations(d.outgoing, false);
+  const causeSets = incoming.sets;
+  const impactSets = outgoing.sets;
+  const nodes = incoming.relNodes.concat(outgoing.relNodes);
+  const links = incoming.relLinks.concat(outgoing.relLinks);
+
+  function parseRelations(ents, isCause = true) {
+    const sets = {};
+    const relNodes = [];
+    const relLinks = [];
+    ents.map((ent) => {
+      const tableName = ent.entityId.dataset;
+      sets[tableName] = [];
+      const fieldIds = new Map();
+      ent.relations.map((rel) => {
+        const fieldName = isCause ? rel.source : rel.destination;
+        let id = fieldIds.get(fieldName);
+        const field = {
+          id,
+          name: fieldName,
+          group: tableName,
+        };
+        if (!fieldIds.has(fieldName)) {
+          id = uuidV4();
```

Comment:
Do we need to have a uuidv4 here? Can the id for a node and corresponding use in links be `tableName_fieldName`? Is there a possibility this can not be unique in this scenario?

Reply:
Based on talking to Andreas, tableName_fieldName is not necessarily unique by itself, but namespace_tableName_fieldName should be unique. I can use that as the id instead, if that seems preferable.

Reply:
That would be good. Trying to avoid using uuidv4 as it is long and may not be performant if we need to generate for all the fields in all datasets.